app/schema.py

```diff
@@ -55,9 +80,95 @@ def mutate(root, info, contact, password):
         return LoginUser(logged_in=True, token=make_hash(user))
 
 
+class SendFriendRequest(graphene.Mutation):
+    class Arguments:
+        contact_receiver = graphene.String()
+    ok = graphene.Boolean()
+    message = graphene.String()
+    user_receiver = graphene.Field(UserType)
+    friend_request = graphene.Field(FriendRequestType)
+
+    @staticmethod
+    @login_required
+    def mutate(root, info, contact_receiver):
+        try:
+            user_receiver = User.objects.get_by_natural_key(contact_receiver)
+
+        except User.DoesNotExist:
+            return SendFriendRequest(message="user not found", ok=False, user_receiver=None, friend_request=None)
+
+        try:
+            friend_request = FriendRequest.objects.get(from_user=info.context.user, to_user=user_receiver)
+            return SendFriendRequest(message="A friend request has already been sent", ok=False, user_receiver=user_receiver, friend_request=None)
+
+        except FriendRequest.DoesNotExist:
+            if FriendRequest.objects.filter(from_user=user_receiver, to_user=info.context.user).exists():
+                return SendFriendRequest(message="User has sent a request before", ok=False, user_receiver=user_receiver, friend_request=None)
+
+            if Friend.objects.filter(current_user=info.context.user, friend=user_receiver).exists():
+                return SendFriendRequest(message="Both users are already friends", ok=False, user_receiver=user_receiver, friend_request=None)
+            
+            friend_request = FriendRequest.objects.create(from_user=info.context.user, to_user=user_receiver)
```

Comment:
You are using `create` method here. But this method does not restrict users from sending friend request to themselves. Please check for this condition and return appropriate message.